Sources/GRPCCodeGen/CodeGenerationRequest.swift

```diff
@@ -282,4 +282,39 @@ public struct CodeGenerationRequest {
       }
     }
   }
+
+  /// Represents the name associated with a namespace, service or a method, in three different formats,
+  /// which are used in specific parts of the generated code. There must be only one ``Name`` object
+  /// for each namespace, service or method.
+  ///
+  /// The base name, the generatedUpperCase (and the generatedLowerCase) must be unique for methods
+  /// from within the same service and for services within the same namespace.
+  public struct Name: Hashable {
+    /// The base name is the name used for the namespace/service/method in the IDL file, so it should follow
+    /// the specific casing of the IDL.
+    ///
+    /// The base name is also used in the descriptors that identify a specific method or service :
+    /// `<service_namespace_baseName>.<service_baseName>.<method_baseName>`.
+    public let base: String
+
+    /// The `generatedUpperCase` name is used as part of generated type names, which begin with a capital letter
+    /// and are (partially) using CamelCase. It is using UpperCamelCase in order to follow the Swift naming conventions.
+    ///
+    /// It is used in the generated code as an enum name and as part of protocol names.
+    /// For example, in the generated server code the name of the service protocol
+    /// follows this pattern:
+    /// `<service_namespace_generatedUpperCaseName>_<service_generatedUpperCaseName>ServiceProtocol`.
```

Comment:
We don't need to be so specific, we're setting ourselves up to have outdated docs (because we may change the pattern and forget about the docs).

In this case the relevant info is that this name is used in generated code and is expected to be the upper-camel-case version of the base name.

You can then provide an abstract example, e.g. if `base` is "echo" then `generatedUpperCase` could be "Echo".